src/routes/CreateNewSafePage/steps/OwnersAndConfirmationsNewSafeStep.tsx

```diff
@@ -0,0 +1,305 @@
+                <Col xs={7}>
+                  <AddressInput
+                    fieldMutator={(address) => {
+                      createSafeForm.change(addressFieldName, address)
+                      const addressName = addressBook[address]?.name
+                      if (addressName) {
+                        createSafeForm.change(nameFieldName, addressName)
+                      }
+                    }}
+                    // eslint-disable-next-line
+                    // @ts-ignore
+                    inputAdornment={
+                      !hasOwnerAddressError && {
+                        endAdornment: (
+                          <InputAdornment position="end">
+                            <CheckCircle
+                              className={classes.checkIcon}
+                              data-testid={`${addressFieldName}-valid-adornment`}
+                            />
+                          </InputAdornment>
+                        ),
+                      }
+                    }
+                    name={addressFieldName}
+                    required
+                    placeholder="Owner Address*"
+                    text="Owner Address"
+                    testId={addressFieldName}
+                  />
+                </Col>
+                <Col xs={1} center="xs" className={classes.ownerIcons} middle="xs">
+                  <ScanQRWrapper handleScan={handleScan} testId={`${addressFieldName}-scan-QR`} />
+                </Col>
+                {showDeleteIcon && (
+                  <Col xs={1} center="xs" className={classes.ownerIcons} middle="xs">
+                    <ButtonHelper
+                      onClick={() => onClickRemoveOwner({ addressFieldName })}
+                      dataTestId={`${addressFieldName}-remove-button`}
+                    >
+                      <Icon size="sm" type="delete" color="icon" tooltip="Delete Owner" />
+                    </ButtonHelper>
+                  </Col>
+                )}
+              </Fragment>
+            )
+          })}
+        </Row>
+        <Row align="center" className={classes.addOwnerContainer} grow>
+          <Button color="secondary" data-testid="add-new-owner" onClick={onClickAddNewOwner}>
+            <Paragraph noMargin size="lg">
+              + Add another owner
+            </Paragraph>
+          </Button>
+        </Row>
+        <div className={classes.padding}>
+          <Block>
+            <Paragraph>Any transaction requires the confirmation of:</Paragraph>
+          </Block>
+          <Row align="center" className={classes.addOwnerContainer} grow>
+            <Col xs={1}>
+              <Field
+                component={SelectField}
+                data-testid="threshold-selector-input"
+                name={FIELD_NEW_SAFE_THRESHOLD}
+                validate={composeValidators(required, minValue(1))}
+              >
+                {Array.from(Array(owners.length).keys()).map((option) => (
```

Comment:
If I understand correctly, you can get the index of the array with the second argument of map:
```ts
owners.map((_, index) =>
```